pallets/file-system/src/lib.rs

```diff
@@ -449,4 +562,40 @@ pub mod pallet {
             total_used_weight
         }
     }
+
+    impl<T: Config> storage_hub_traits::SubscribeProvidersInterface for Pallet<T> {
+        type Provider = StorageProviderId<T>;
+
+        fn subscribe_bsp_sign_up(_who: &Self::Provider) {
+            // Adjust bsp assignment threshold by applying the decay function after removing the asymptote
+            let mut bsp_assignment_threshold = BspsAssignmentThreshold::<T>::get();
+            let base_threshold = bsp_assignment_threshold
+                .checked_sub(&T::AssignmentThresholdAsymptote::get())
+                .unwrap_or_else(|| panic!("Underflow in threshold calculation during sign-up"));
```

Comment:
We should avoid this `panic`s in the runtime. Use `expect_or_else` instead. Besides, using `unwrap_or_else` and then panicking is kinda redundant. It is equivalent to using just `expect()`

Reply:
Return `DispatchResult` now with `ThresholdArithmeticError`.